### PR TITLE
chore: remaining FVM pruning

### PR DESCRIPTION
**Pull request by jdjaustin**
## Summary of changes

Changes introduced in this pull request:

- Replace direct `fvm_shared` and `fvm_shared3` usages with implementations in `forest_shim`

## Reference issue to close (if applicable)

Closes associated tasks in #2962 

## Other information and links

The banned dependencies script will be updated by PR #3110.

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/shim/crypto.rs [hunk 1]
```diff
@@ -149,6 +154,49 @@ pub fn verify_bls_aggregate(data: &[&[u8]], pub_keys: &[&[u8]], sig: &Signature)
     verify_messages(&bls_sig, data, &pks[..])
 }
 
+/// Returns `String` error if a BLS signature is invalid.
+pub fn verify_bls_sig(signature: &[u8], data: &[u8], addr: &Address) -> Result<(), String> {
```
Comment (lemmih) [inline src/shim/crypto.rs:158]: Why have you copied this code?

File: src/shim/crypto.rs [hunk 2]
```diff
@@ -149,6 +154,49 @@ pub fn verify_bls_aggregate(data: &[&[u8]], pub_keys: &[&[u8]], sig: &Signature)
...
+            "bls signature verification failed for addr: {}",
+            addr
+        ))
+    }
+}
+
+/// Extracts the raw replica commitment from a CID
+/// assuming that it has the correct hashing function and
+/// serialization types
+pub fn cid_to_replica_commitment_v1(c: &Cid) -> Result<Commitment, &'static str> {
```
Comment (lemmih) [inline src/shim/crypto.rs:190]: Why have you copied this code?

File: src/state_migration/nv17/tests/miner_tests.rs [hunk 3]
```diff
@@ -0,0 +1,571 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#[cfg(test)]
+mod tests {
```
Comment (lemmih) [inline src/state_migration/nv17/tests/miner_tests.rs:5]: Why have you moved the tests here?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: src/state_migration/nv17/tests/miner_tests.rs (hunk 3 above)
Comment (jdjaustin) [reply src/state_migration/nv17/tests/miner_tests.rs:5]: To create an exception for `'src/state_migration/nv17/tests/'` when checking banned dependencies under PR #3110.

File: src/shim/crypto.rs (hunk 1 above)
Comment (jdjaustin) [reply src/shim/crypto.rs:158]: To replace the function from FVM for usage in `src/fil_cns/validation.rs`; I suppose I could have invoked the function without copying the code, but I was being consistent with the `verify_bls_aggregate` function above this.

File: src/shim/crypto.rs (hunk 2 above)
Comment (jdjaustin) [reply src/shim/crypto.rs:190]: See reply above for `verify_bls_sig`.

File: src/shim/crypto.rs (hunk 1 above)
Comment (lemmih) [reply src/shim/crypto.rs:158]: Our copy of `verify_bls_aggregate` is semantically different from `fvm_shared::crypto::signature::verify_bls_aggregate`. If you accidentally used the FVM function, Forest would stop working.

Your copy of `verify_bls_sig` is not different from the FVM version. So let's use the FVM version.

File: src/state_migration/nv17/tests/miner_tests.rs (hunk 3 above)
Comment (lemmih) [reply src/state_migration/nv17/tests/miner_tests.rs:5]: Let's change the script rather than change the code.

**Review by lemmih — APPROVED** [review]

**Pull request merged**